From e549e303cfa9dabdb9c39ede6c5b2f5936269aab Mon Sep 17 00:00:00 2001
From: Maximiliano Sandoval <msandova@protonmail.com>
Date: Sun, 3 Mar 2024 16:31:54 +0100
Subject: [PATCH] appdata: Improvements for AppStream 1.0

- Add developer
- Add missing releases
- Add launchable
- Use secure URI for arduino.cc
---
 build/linux/dist/appdata.xml | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/build/linux/dist/appdata.xml b/build/linux/dist/appdata.xml
index 9f8b32499..ba06ff8f9 100644
--- a/build/linux/dist/appdata.xml
+++ b/build/linux/dist/appdata.xml
@@ -5,6 +5,9 @@
   <metadata_license>CC-BY-SA-3.0</metadata_license>
   <project_license>LGPL-2.1</project_license>
   <developer_name>Arduino LLC</developer_name>
+  <developer id="cc.arduino">
+    <name>Arduino LLC</name>
+  </developer>
 
   <name>Arduino IDE</name>
   <summary>Open-source electronics prototyping platform</summary>
@@ -36,6 +39,19 @@
   <content_rating type="oars-1.1" />
 
   <releases>
+    <release date="2021-12-16" version="1.8.19">
+      <description>
+        <ul>
+          <li>Removed log4j from IDE. Fix CVE-2021-45105, CVE-2021-45046</li>
+        </ul>
+      </description>
+    </release>
+    <release date="2021-12-16" version="1.8.18"/>
+    <release date="2021-09-27" version="1.8.16"/>
+    <release date="2021-05-14" version="1.8.15"/>
+    <release date="2020-05-12" version="1.8.14"/>
+    <release date="2020-06-16" version="1.8.13"/>
+    <release date="2020-02-13" version="1.8.12"/>
     <release date="2020-01-27" version="1.8.11"/>
     <release date="2019-09-13" version="1.8.10"/>
     <release date="2019-03-15" version="1.8.9"/>
@@ -49,11 +65,12 @@
     <release date="2017-01-09" version="1.8.1"/>
   </releases>
 
-  <url type="homepage">http://www.arduino.cc/</url>
+  <url type="homepage">https://www.arduino.cc/</url>
   <url type="help">https://www.arduino.cc/en/Guide/HomePage</url>
   <url type="bugtracker">https://github.com/arduino/Arduino/issues</url>
   <url type="translate">https://github.com/arduino/Arduino/tree/master/arduino-core/src/processing/app/i18n</url>
   <url type="donation">https://www.arduino.cc/en/Main/Contribute</url>
+  <launchable type="desktop-id">arduino.desktop</launchable>
 
   <update_contact>arduino.appdata.xml@mavit.org.uk</update_contact>
 </component>
-- 
2.44.0

